### ISSUE TITLE
Bloop compiler crashes on compilation error

### ISSUE THREAD

**Issue by alexfoxgill** [scalacenter/bloop#1474]
When running `bloop compile root --verbose` on one particular project (unfortunately a work project that i can't share) I see an error with this stack trace:
https://gist.github.com/alexfoxgill/b55773b096a425230b0265f0a934a5dd

**Comment by tgodzik** [comment]
Thanks for reporting! Looking at the issue it seems that we are rethrowing the exception on purpose :thinking:

**Comment by jozanek** [comment]
This is an upstream `scalac` pattern-matcher bug, not Bloop — the whole stack above `xsbt.CachedCompiler0.run` is inside the compiler's `patmat` phase. Bloop just surfaces it.

The line numbers pin your compiler to **Scala 2.12.12 / 2.12.13** (`MatchTranslation.scala:222` = `repeatedToSeq(origPt)` only matches those). The NPE means a synthetic `Match` tree has a `null` type, which blows up in `elementExtract`. Same signature as [scala/bug#10483](https://github.com/scala/bug/issues/10483) and [scala/bug#8337](https://github.com/scala/bug/issues/8337) — neither was minimizable, and I couldn't reproduce it standalone either.

Re: the rethrow — that's intentional. Unexpected compiler exceptions are surfaced on purpose, otherwise the compile task hangs forever. So Bloop is working as designed.

To get this fixed upstream we'd need a repro. If you can share a reduced version of the offending source, `scalac -Ystop-after:patmat -Vprint:patmat YourFile.scala` will show the last match before the crash — bisect from there. (Upgrading Scala may help, but the vulnerable path is still unguarded as of 2.13.18, so no guarantee.)

**Comment by tgodzik** [comment]
Yeah, makes sense to close it if it's a compiler issue. We should throw indeed.
